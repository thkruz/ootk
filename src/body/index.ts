export { Celestial } from './Celestial';
export { Earth } from './Earth';
export { Moon } from './Moon';
export { Sun } from './Sun';
