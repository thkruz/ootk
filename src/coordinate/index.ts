export * from './Hill.js';
export * from './ClassicalElements.js';
export * from './EquinoctialElements.js';
export * from './FormatTle.js';
export * from './Geodetic.js';
export * from './ITRF.js';
export * from './J2000.js';
export * from './RelativeState.js';
export * from './RIC.js';
export * from './StateVector.js';
export * from './TEME.js';
export * from './Tle.js';
