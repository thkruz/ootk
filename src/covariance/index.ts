export * from './CovarianceSample.js';
export * from './StateCovariance.js';
