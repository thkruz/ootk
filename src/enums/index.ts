export { CatalogSource } from './CatalogSource.js';
export { CommLink } from './CommLink.js';
export { OrbitRegime } from './OrbitRegime.js';
export { Sgp4OpsMode } from './Sgp4OpsMode.js';
export { AngularDiameterMethod } from './AngularDiameterMethod.js';
export { AngularDistanceMethod } from './AngularDistanceMethod.js';
export { PassType } from './PassType.js';
