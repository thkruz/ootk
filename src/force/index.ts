export { EarthGravity } from './EarthGravity.js';
export { Force } from './Force.js';
export { ForceModel } from './ForceModel.js';
export { Gravity } from './Gravity.js';
export { SolarRadiationPressure } from './SolarRadiationPressure.js';
export { ThirdBodyGravity } from './ThirdBodyGravity.js';
export { Thrust } from './Thrust.js';

