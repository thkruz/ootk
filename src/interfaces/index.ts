export type { BaseObjectParams } from './BaseObjectParams.js';
export type { GroundPositionParams } from './GroundPositionParams.js';
export type { OptionsParams } from './OptionsParams.js';
export type { SatelliteParams } from './SatelliteParams.js';
export type { SensorParams } from './SensorParams.js';
export type { StarObjectParams } from './StarObjectParams.js';
export type { ClassicalElementsParams } from './ClassicalElementsParams.js';
export type { EquinoctialElementsParams } from './EquinoctialElementsParams.js';
export type { RfSensorParams } from './RfSensorParams.js';
