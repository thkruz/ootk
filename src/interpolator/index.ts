export { ChebyshevCoefficients } from './ChebyshevCoefficients.js';
export { ChebyshevInterpolator } from './ChebyshevInterpolator.js';
export { CubicSpline } from './CubicSpline.js';
export { CubicSplineInterpolator } from './CubicSplineInterpolator.js';
export { Interpolator } from './Interpolator.js';
export { LagrangeInterpolator } from './LagrangeInterpolator.js';
export { StateInterpolator } from './StateInterpolator.js';
export { VerletBlendInterpolator } from './VerletBlendInterpolator.js';

