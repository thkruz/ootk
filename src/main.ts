/**
 * @author @thkruz Theodore Kruczek
 * @description Orbital Object ToolKit (ootk) is a collection of tools for working
 * with satellites and other orbital objects.
 * @license AGPL-3.0-or-later
 * @copyright (c) 2020-2024 Theodore Kruczek
 *
 * Many of the classes are based off of the work of @david-rc-dayton and his
 * Pious Squid library (https://github.com/david-rc-dayton/pious_squid).
 *
 * Orbital Object ToolKit is free software: you can redistribute it and/or modify it under the
 * terms of the GNU Affero General Public License as published by the Free Software
 * Foundation, either version 3 of the License, or (at your option) any later version.
 *
 * Orbital Object ToolKit is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
 * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * See the GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License along with
 * Orbital Object ToolKit. If not, see <http://www.gnu.org/licenses/>.
 */

export * from './enums/index.js';
export * from './types/types.js';
export * from './interfaces/index.js';
export * from './time/index.js';
export * from './transforms/index.js';
export * from './utils/index.js';
export * from './operations/operations.js';
export { BaseObject, Sensor } from './objects/index.js';
export { Earth } from './body/index.js';
export * from './coordinate/index.js';
export * from './observation/index.js';
export * from './data/DataHandler.js';
export * from './sgp4/index.js';
export * from './objects/index.js';
export * from './interpolator/index.js';
export * from './force/index.js';
export * from './maneuver/index.js';
export * from './optimize/index.js';
export * from './orbit_determination/index.js';
export * from './body/index.js';
export * from './enums/index.js';
export * from './types/types.js';
export * from './interfaces/index.js';
export * from './coordinate/index.js';
export * from './operations/index.js';
export * from './force/index.js';
export * from './propagator/index.js';
export * from './transforms/index.js';
export * from './utils/index.js';
