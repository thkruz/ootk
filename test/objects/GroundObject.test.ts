import { Degrees, Geodetic, GroundObject, Kilometers, Radians } from '../../src/main';

describe('GroundObject', () => {
  let groundObject: GroundObject;
  let geodetic: Geodetic;
  let exampleDate: Date;

  beforeEach(() => {
    const mockExampleDate = new Date(1705109326817);

    exampleDate = new Date(mockExampleDate.getTime());
    groundObject = new GroundObject({
      lat: 0 as Degrees,
      lon: 0 as Degrees,
      alt: 0 as Kilometers,
    });

    geodetic = new Geodetic(
      0 as Radians,
      0 as Radians,
      0 as Kilometers,
    );
  });

  it('should be defined', () => {
    expect(groundObject).toBeDefined();
  });

  it('should have a name', () => {
    expect(groundObject.name).toBeDefined();
  });

  it('should have a latitude', () => {
    expect(groundObject.lat).toBeDefined();
  });

  it('should have a longitude', () => {
    expect(groundObject.lon).toBeDefined();
  });

  it('should calculate ecf', () => {
    expect(groundObject.ecf()).toMatchSnapshot();
  });

  it('should calculate eci', () => {
    expect(groundObject.eci(exampleDate)).toMatchSnapshot();
  });

  it('should calculate llaRad', () => {
    expect(groundObject.llaRad()).toMatchSnapshot();
  });

  it('should create from geodetic', () => {
    expect(GroundObject.fromGeodetic(geodetic)).toMatchSnapshot();
  });

  it('should convert to geodetic', () => {
    expect(groundObject.toGeodetic()).toMatchSnapshot();
  });
});
