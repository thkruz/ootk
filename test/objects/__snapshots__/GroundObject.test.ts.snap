// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroundObject should calculate ecf 1`] = `
Object {
  "x": 6378.137,
  "y": 0,
  "z": 0,
}
`;

exports[`GroundObject should calculate eci 1`] = `
Object {
  "x": -4444.532338535956,
  "y": 4564.633188736234,
  "z": 0,
}
`;

exports[`GroundObject should calculate llaRad 1`] = `
Object {
  "alt": 0,
  "lat": 0,
  "lon": 0,
}
`;

exports[`GroundObject should convert to geodetic 1`] = `
Geodetic {
  "alt": 0,
  "lat": 0,
  "lon": 0,
}
`;

exports[`GroundObject should create from geodetic 1`] = `
GroundObject {
  "active": true,
  "alt": 0,
  "id": -1,
  "lat": 0,
  "lon": 0,
  "name": "Unknown Ground Object",
  "position": Object {
    "x": 0,
    "y": 0,
    "z": 0,
  },
  "totalVelocity": 0,
  "type": 0,
  "velocity": Object {
    "x": 0,
    "y": 0,
    "z": 0,
  },
}
`;
